pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -308,24 +308,50 @@ type VitessShardStatus struct {
 	// HasMaster is a condition indicating whether the Vitess topology
 	// reflects a master for this shard.
 	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
+
 	// HasInitialBackup is a condition indicating whether the initial backup
 	// has been seeded for the shard.
 	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
+
 	// MasterAlias is the tablet alias of the master according to the global
 	// shard record. This could be empty either because there is no master,
 	// or because the shard record could not be read. Check the HasMaster
 	// condition whenever the distinction is important.
 	MasterAlias string `json:"masterAlias,omitempty"`
 
+	// BackupLocations reports information about the backups for this shard in
+	// each backup location.
+	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+
 	// Idle is a condition indicating whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
-	// BackupLocations reports information about the backups for this shard in
-	// each backup location.
-	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
+}
+
+// VitessShardConditionType is a valid value for VitessShardCondition.Type
+type VitessShardConditionType string
+
+// VitessShardCondition contains details for the current condition of this VitessShard.
+type VitessShardCondition struct {
+	// Type is the type of the condition.
+	Type VitessShardConditionType `json:"type"`
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
```

Comment:
These `+optional` tags also don't do anything, I believe, since we don't run the same tooling as core k8s. Let's remove them to avoid the possibility of getting out of sync.

Reply:
In our codebase, I've often seen `Optional.` on its own line. Do you think it would be useful to change to that?

Reply:
sgtm

Reply:
Done https://github.com/planetscale/vitess-operator/pull/31/commits/bf051469919e3a3e7686f689c0766983e7852b11.